### DISCUSSION TITLE
Please add an auto-delete function after password fail attempts

### DISCUSSION THREAD

**Discussion by seularts** [keepassxreboot/keepassxc#8505] [Ideas]
Before using KeePassXC I used to have this stand-alone password manager app just for the phone that had this simple optional function of auto-delete the database after x amount of fail login attempts. I think this should be here as well as it is a good way of preventing someone from bruteforcing your database! As an extra idea: you could add an additional option to hide the number of attempts so the individual doesn't realize he is about to auto-delete everything. I know that some smart would make copies before attempting this, but it's a nice way of having some more security!

**Comment by michaelk83** [comment]
I can already see the complaints from the users who didn't backup their database, and mistyped the passphrase a few times, and OOPS!
This gives you exactly *zero* extra security, since taking a backup first is trivial, and any competent *user* - let alone an attacker - would always have at least one backup. If you want to protect yourself against brute-force attacks, just use a long enough passphrase. The algorithms are already designed to make brute-forcing impractical if the passphrase is strong enough.
This is a terrible idea.

**Reply by seularts** [reply]
I was referring to my idea as an optional setting, not a mandatory one that comes activated by default. And not all attackers are tech-smart, could just be a jealous spouse that wants to guess the master password. You are right, a smart user always has his database backed up. What I am asking is for a fail-safe against close people who could snatch your device and try to do their devious guessing work on it. Also, the user should be able to set up how many times a password must be entered incorrectly before the db is erased.

**Reply by michaelk83** [reply]
Do you really want all your passwords to be gone as soon as someone makes a few *unsuccessful* attempts at accessing them? This is a terrible idea even as an optional setting.

Just to make things clearer: the encryption algorithms are tuned by default to [take 1 sec per decryption attempt](https://keepassxc.org/docs/KeePassXC_UserGuide.html#_creating_your_first_database) (scroll down a bit in that link to see the encryption settings). This is small for a legitimate user, but eternity for an attacker. I you use a strong enough password, it would statistically take many billions of attempts to find the right one by brute force. At 1 sec per attempt, that would take decades for a moderately strong passphrase, and possibly more than a millenium for a strong one. Not practical even for a sophisticated attacker.
(Of course, they *could* try to decrypt on a much stronger machine, but even then it would take them many years at best, and why would they waste all those time and resources on *your* passwords?)

Your curious wife would simply give up after a few attempts. She will most likely fail, and your passwords will still be safe. There's no need to delete the file.

**Reply by michaelk83** [reply]
Let's say you use the built-in passphrase generator to generate a passphrase. The default word list for this generator has over 7000 entries. Just using 3 random words from there would give over 300 billion possibilities. 4 words gives over a million billion (10^15). You should use at least 4 words.

**Reply by seularts** [reply]
If my device would be taken by a spouse, or even a friend that doesn't admit taking it, yes I would like them to lose access to that database (personally I do my db backups externally from these devices for safe keeping). They could use more rudimentary ways of observing my password, for example by seeing me type it when I am not paying much attention to them. Clearly, the algorithm would be extremely difficult to break since I use long passwords, but then again I have typed my password next to friends via my phone or laptop, and you know.. you can never be too cautious in this day and age and changing the password every time someone is next to me would be borderline paranoia and impractical (at least for me) 🗡️

**Reply by michaelk83** [reply]
If you're really worried about someone you know accessing your passwords, don't type your passphrase next to them, and use a YubiKey as a 2nd factor. And if you really, *really* can't trust them, you may want to reconsider your entire relationship.

**Reply by seularts** [reply]
Or simply have the auto-destruct db that solves all of my problems :D For now I do not have the Yubikey. It's just a feature I am asking for, the devs could simply say no and I will move on with my life.

**Reply by michaelk83** [reply]
A more sensible option would be to forbid further login attempts for the next 24 hours or so. But that's not particularly useful for most users, either.

Bottom line is - if your spouse/friends know your passphrase, even a self-destruct won't help you; if they don't, your passwords will still be safe without deleting the file. But if you've surrounded yourself with people who don't respect your privacy and personal space, or that you feel like you need to hide things from (especially your spouse), you have far bigger problems than your passwords database.

**Reply by seularts** [reply]
Auto-deleting my nosy friends is not an option. But yeah, it was just an idea I threw in the ring, please don't hate me for it :)

**Reply by michaelk83** [reply]
No hate intended.

**Reply by droidmonkey** [reply]
> the devs could simply say no and I will move on with my life.

No

**Reply by seularts** [reply]
As promised devs, I am moving on with my life 🍡

**Comment by phoerious** [comment]
This protects against nothing. The difficult part is getting access to your computer and whoever is successful in that will not be trying to bruteforce your database on your computer by entering a billion password attempts by hand.
